### PR TITLE
Adds user -> account linking

### PR DESCRIPTION
**Pull request by michaloo**
* fix the `requestExtract` handler - allow passing `path` param
* fix the `.asUser()` and `.asAccount()` to return `traits` and `track`
* adds `.asUser().account()` method

### CODE REVIEW

**Comment by michaloo** [comment]
[Accounts JS API](https://trello.com/c/Q7YycVOT/188-accounts-js-api)

File: tests/client-tests.js [hunk 1]
```diff
@@ -50,26 +67,62 @@ describe("Hull", () => {
...
-      const scopedJwtClaims = jwt.decode(scopedConfig.accessToken, scopedConfig.secret);
+      const scoped = hull.asUser({ email: "foo@bar.com" }).account({ name: "Hull" });
+      const scopedJwtClaims = jwt.decode(scoped.configuration().accessToken, scoped.configuration().secret);
+
+      expect(scopedJwtClaims)
+        .to.have.property("io.hull.subjectType")
+        .that.eql("account");
       expect(scopedJwtClaims)
         .to.have.property("io.hull.asAccount")
         .that.eql({ name: "Hull" });
```
Comment (KgaevskiT) [inline tests/client-tests.js]: Again, `name` is not available here

File: tests/client-tests.js [hunk 2]
```diff
@@ -50,26 +67,62 @@ describe("Hull", () => {
...
+      expect(scopedJwtClaims)
+        .to.have.property("io.hull.asUser")
+        .that.eql({ email: "foo@bar.com" });
     });
+
+    it("should allow to link user to an existing account", () => {
```
Comment (KgaevskiT) [inline tests/client-tests.js]: This is not actually linking. It is just a getter to the user's account

File: tests/client-tests.js [hunk 3]
```diff
@@ -50,26 +67,62 @@ describe("Hull", () => {
         .that.eql("123456");
     });
 
-    it("should allow to pass account id as an object property", () => {
+    it("should allow to pass account name as an object property", () => {
       const hull = new Hull({ id: "562123b470df84b740000042", secret: "1234", organization: "test" });
 
-      const scoped = hull.asAccount({ id: "123456" });
+      const scoped = hull.asAccount({ name: "Hull" });
```
Comment (michaloo) [inline tests/client-tests.js]: Okey, I've removed it from accountClaim properties and adjusted tests.

File: tests/client-tests.js (hunk 2 above)
Comment (michaloo) [inline tests/client-tests.js]: You're right, fixed.

File: tests/client-tests.js [hunk 4]
```diff
@@ -50,26 +67,62 @@ describe("Hull", () => {
...
       const scopedConfig = scoped.configuration();
       const scopedJwtClaims = jwt.decode(scopedConfig.accessToken, scopedConfig.secret);
       expect(scopedJwtClaims)
-        .to.have.property("sub")
-        .that.eql("123456");
+        .to.have.property("io.hull.asAccount")
+        .that.eql({ name: "Hull" });
     });
```
Comment (michaloo) [inline tests/client-tests.js]: We have one test for `hull.asAccount()` without user resolution. I would test `traits` on other place, but thanks, for pointing it out.

**Pull request merged**